src/Build/Graph/ParallelWorkSet.cs

```diff
@@ -67,7 +61,11 @@ internal class ParallelWorkSet<TKey, TResult>
         /// <summary>
         /// Checks if the work set has been marked as completed.
         /// </summary>
-        internal bool IsCompleted => _isSchedulingCompleted;
+        internal bool IsCompleted
+        {
+            get => Volatile.Read(ref _isSchedulingCompleted);
+            set => Volatile.Write(ref _isSchedulingCompleted, value);
```

Comment:
Why is this public API surface? Is it meaningful for a caller to set it?
```suggestion
            private set => Volatile.Write(ref _isSchedulingCompleted, value);
```